layers/terrain-material.js

```diff
@@ -25,200 +19,300 @@ const _loadTexture = async (path) => {
   return texture;
 };
 
-const loadTerrainMaterial = async () => {
-  const rockDiffMap = await _loadKTX2(
-    'assets/textures/rock/rock_06_diff_8k.ktx2'
-  );
-  rockDiffMap.encoding = THREE.sRGBEncoding;
-
-  const rockRoughnessMap = await _loadKTX2(
-    'assets/textures/rock/rock_06_rough_8k.ktx2'
-  );
-  rockRoughnessMap.encoding = THREE.LinearEncoding;
-
-  const rockNormalMap = await _loadKTX2(
-    'assets/textures/rock/rock_06_nor_dx_8k.ktx2'
-  );
-  rockNormalMap.encoding = THREE.LinearEncoding;
-
-  const rockAoMap = await _loadKTX2(
-    'assets/textures/rock/rock_06_nor_dx_8k.ktx2'
-  );
-  rockNormalMap.encoding = THREE.LinearEncoding;
-
-  const noiseTexture = await _loadTexture('assets/textures/simplex-noise.png');
-  noiseTexture.encoding = THREE.LinearEncoding;
 
-  // define material uniforms here
+const DIFFUSE_PATHS = [
+  baseUrl + '../assets/textures/stylized_grass/stylized_grass1_d.png',
+  baseUrl + '../assets/textures/dirt/dirt1_d.png'
+];
+
+const NORMAL_PATHS = [
+  baseUrl + '../assets/textures/stylized_grass/stylized_grass1_n.png',
+  baseUrl + '../assets/textures/dirt/dirt1_n.png',
+];
+
+export const NUM_MATERIALS = DIFFUSE_PATHS.length; // TODO : get this number from wasm
+
+const loadTerrainMaterial = async () => {
```

Comment:
I would separate the loading, baking, and material into separate structured parts of code.

See how it's done with the package concept:
- https://github.com/webaverse/heightfield/blob/0db67c9e60589a8d019b5c82cf38855db05a2d0b/meshes/icon-mesh.js#L41
- https://github.com/webaverse/heightfield/blob/0db67c9e60589a8d019b5c82cf38855db05a2d0b/meshes/polygon-mesh.js#L27
- https://github.com/webaverse/heightfield/blob/0db67c9e60589a8d019b5c82cf38855db05a2d0b/meshes/spritesheet-mesh.js#L25

Usage:
- https://github.com/webaverse/heightfield/blob/7e8ebb0ed25af4ae0fe102b461bafcc96b7ede36/layers/litter-mesh.js#L79